sql/core/src/test/scala/org/apache/spark/sql/JoinSuite.scala

```diff
@@ -215,18 +215,36 @@ class JoinSuite extends QueryTest with SharedSQLContext {
           Row(1, null, 2, 2) ::
           Row(2, 2, 1, null) ::
           Row(2, 2, 2, 2) :: Nil)
+
+      checkAnswer(
+        testData3.as("x").join(testData3.as("y"), $"x.a" > $"y.a"),
```

Comment:
This is a typical example of cartesian product, but our current detection is unable to find.